src/Workleap.Authentication.ClientCredentialsGrant.Tests/IntegrationTests.cs

```diff
@@ -1,278 +1,156 @@
-            var tokenAfterSecondBackgroundRefresh = await tokenCache.GetAsync(invoiceReadClientName, cts.Token);
-            Assert.NotNull(tokenAfterSecondBackgroundRefresh);
-            Assert.NotEqual(tokenAfterFirstBackgroundRefresh, tokenAfterSecondBackgroundRefresh);
-        }
-        finally
-        {
-            // Shut down the web app
-#pragma warning disable CA1849 // Given that we are using .Net6/7/8, updating the method to CancelAsync would break backwards compatibility
-            cts.Cancel();
-#pragma warning restore CA1849
-        }
+        // TODO improve background token acquisition so that 2nd app uses the token cached by the 1st app from the distributed cache
+        Assert.Equal(2, this._tokenRequestCount);
     }
 
-    private sealed class TestServerHandler : DelegatingHandler
+    [Fact]
+    public async Task GivenAnonymousEndpoint_WhenReadAuthenticatedHttpRequest_ShouldBeSuccessful()
     {
-        private readonly TestServer _testServer;
-        private HttpClient? _testServerClient;
-
-        public TestServerHandler(TestServer testServer)
-        {
-            this._testServer = testServer;
-        }
-
-        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
-        {
-            this._testServerClient ??= this._testServer.CreateClient();
-
-            // Request has to be cloned since it has already gone through the httpclient wrapping this handler even though the request hasn't been sent yet.
-            var cloneRequest = await CloneHttpRequest(request, cancellationToken);
-
-            return await this._testServerClient.SendAsync(cloneRequest, cancellationToken);
-        }
-
-        private static async Task<HttpRequestMessage> CloneHttpRequest(HttpRequestMessage request, CancellationToken cancellationToken)
-        {
-            var cloneRequest = new HttpRequestMessage(request.Method, request.RequestUri);
-            cloneRequest.Version = request.Version;
+        var httpClient = this._api1.Services.GetRequiredService<IHttpClientFactory>().CreateClient(InvoiceReadHttpClientName);
+        var response = await httpClient.GetStringAsync("https://invoice-app.local/anonymous", this._timeout.Token);
+        Assert.Equal("This endpoint is public", response);
+    }
 
-            foreach (var (key, value) in request.Headers)
-            {
-                cloneRequest.Headers.TryAddWithoutValidation(key, value);
-            }
+    [Fact]
+    public async Task GivenReadEndpointWithClassicPolicy_WhenReadAuthenticatedHttpRequest_ShouldBeSuccessful()
+    {
+        var httpClient = this._api1.Services.GetRequiredService<IHttpClientFactory>().CreateClient(InvoiceReadHttpClientName);
+        var response = await httpClient.GetStringAsync("https://invoice-app.local/read-invoices", this._timeout.Token);
+        Assert.Equal("This protected endpoint is for reading invoices", response);
```

Comment:
Would the test be less fragile if instead of comparing the content that is defined in an other file we simply check for the status code?